From 17a11b2640901eafe863707645088b4907d18304 Mon Sep 17 00:00:00 2001
From: Dinesh Karthikeyan <quic_dinek@quicinc.com>
Date: Fri, 21 Apr 2023 13:50:03 +0530
Subject: [PATCH] ath12k: Modify the M3 SSR Dump address

Add changes to fetch the M3 dump address for IPQ5322 from
dtsi.

Signed-off-by: Dinesh Karthikeyan <quic_dinek@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/qmi.c | 14 ++++++++++++--
 1 file changed, 12 insertions(+), 2 deletions(-)

diff --git a/drivers/net/wireless/ath/ath12k/qmi.c b/drivers/net/wireless/ath/ath12k/qmi.c
index 302edfd..4b0f576 100644
--- a/drivers/net/wireless/ath/ath12k/qmi.c
+++ b/drivers/net/wireless/ath/ath12k/qmi.c
@@ -3973,8 +3973,8 @@ static int ath12k_qmi_assign_target_mem_chunk(struct ath12k_base *ab)
 {
 	struct device *dev = ab->dev;
 	struct ath12k_hw_group *ag = ab->ag;
-	struct device_node *hremote_node = NULL, *mem_node;
-	struct resource res;
+	struct device_node *hremote_node = NULL, *mem_node, *dev_node = NULL;
+	struct resource res, m3_dump;
 	int host_ddr_sz, mlo_ddr_sz, sz, mlo_sz = 0;
 	int i, idx, mlo_idx, ret;
 	unsigned int bdf_location[MAX_TGT_MEM_MODES],
@@ -4115,6 +4115,16 @@ skip_mlo_mem_init:
 			idx++;
 			break;
 		case M3_DUMP_REGION_TYPE:
+			if (ab->hif.bus == ATH12K_BUS_AHB) {
+				dev_node = of_find_node_by_name(NULL, "m3_dump");
+				if (of_address_to_resource(dev_node, 0, &m3_dump)) {
+					ath12k_err(ab, "M3_MEM_REGION Not defined in device_tree\n");
+					ret = -EINVAL;
+					goto out;
+				} else {
+					ab->qmi.target_mem[idx].paddr = m3_dump.start;
+				}
+			}
                         ab->qmi.target_mem[idx].v.ioaddr =
                                         ioremap(ab->qmi.target_mem[idx].paddr,
 						ab->qmi.target_mem[i].size);
-- 
2.17.1

